ctk-transport/src/main/java/org/ga4gh/ctk/transport/protocols/Client.java

```diff
@@ -112,16 +126,14 @@ public SearchDatasetsResponse searchDatasets(SearchDatasetsRequest request)
          * <tt>GET /datasets/{id}</tt> returns a {@link Dataset}.
          *
          * @param id the ID of the dataset
-         * @throws AvroRemoteException if there's a communication problem
+         * @throws GAWrapperException if the server finds the request invalid in some way
+         * @throws UnirestException if there's a problem speaking HTTP to the server
+         * @throws InvalidProtocolBufferException if there's a problem processing the JSON response from the server
          */
-        @Override
-        public Dataset getDataset(String id) throws AvroRemoteException {
-            String path = urls.getGetDataset();
-            Dataset response = new Dataset();
-            final AvroJson aj =
-                    new AvroJson<>(response, urls.getUrlRoot(), path, wireTracker);
-            response = (Dataset) aj.doGetResp(id);
-            return response;
+        public Dataset getDataset(String id) throws InvalidProtocolBufferException, UnirestException, GAWrapperException {
+            Dataset.Builder responseBuilder = Dataset.newBuilder();
+            new Get<>(urls.getUrlRoot(), urls.getGetDataSet(), id, null, responseBuilder, wireTracker).performQuery();
```

Comment:
Discussed IRL. Basically yes - the Base captures the entire lifecycle of the request, whether it's a Get or a Post.